docs/hardware/dbg.md

```diff
@@ -0,0 +1,51 @@
+# Debug connector
+
+It's often needed to connect a SWD debugger to a board.
+
+We found [Black Magic Probe](https://github.com/blacksphere/blackmagic/wiki) to work well,
+for a variety of MCUs, and it doesn't require openocd.
+You can use their beautifully made hardware, or flash an existing
+board with Black Magic firmware.
+People also often use ST Link v2 clones, especially for STM chips.
+
+One way or another, you need to connect the SWDIO and SWCLK lines from the
+debugger to the MCU.
+Often these are connected to test pins, to which you have to solder.
+Sometimes there is a standard 0.05" 10 pin ARM debug connector,
+however these require a part to be soldered on the board.
+Otherwise, people use a footprint on the PCB to which they connect a 
+special proprietary (and expensive) connector cable.
+
+Here we propose simple and cost effective solutions, which do not require soldering.
+
+## One row connector
+
+Use a standard 0.01" spacing header holes with 40mil drill and place them so that
+the middle of the holes are exactly 0.01" from the edge of the board.
+Then use a two-row male header, and plug it into the holes on the short side
+and connect female jumper cables on the long side.
+The header will bend a little, forcing good electrical connection.
+You usually want to plug it starting from the edge.
+
+![Debug connector](/static/hardware/dbg.jpg)
```

Comment:
Image doesn't show!

Reply:
it will in docs; you can click on "view file" at the bottom to see it